main.go

```diff
@@ -447,3 +428,109 @@ func handleTriggerUninstall() {
 		os.Exit(1)
 	}
 }
+
+func setupOpts(flags *pflag.FlagSet) *ctrlOpts {
+	opts := &ctrlOpts{}
+
+	flags.UintVar(
+		&opts.frequency,
+		"update-frequency",
+		10,
+		"The status update frequency (in seconds) of a mutation policy",
+	)
+
+	flags.BoolVar(
+		&opts.enableLease,
+		"enable-lease",
+		false,
+		"If enabled, the controller will start the lease controller to report its status",
+	)
+
+	flags.StringVar(
+		&opts.clusterName,
+		"cluster-name",
+		"acm-managed-cluster",
+		"Name of the cluster",
+	)
+
+	flags.StringVar(
+		&opts.hubConfigPath,
+		"hub-kubeconfig-path",
+		"/var/run/klusterlet/kubeconfig",
+		"Path to the hub kubeconfig",
+	)
+
+	flags.StringVar(
+		&opts.targetKubeConfig,
+		"target-kubeconfig-path",
+		"",
+		"A path to an alternative kubeconfig for policy evaluation and enforcement.",
+	)
+
+	flags.StringVar(
+		&opts.metricsAddr,
+		"metrics-bind-address",
+		"localhost:8383",
+		"The address the metrics endpoint binds to.",
+	)
+
+	flags.StringVar(
+		&opts.probeAddr,
+		"health-probe-bind-address",
+		":8081",
+		"The address the probe endpoint binds to.",
+	)
+
+	flags.BoolVar(
+		&opts.enableLeaderElection,
+		"leader-elect",
+		true,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.",
+	)
+
+	flags.BoolVar(
+		&opts.legacyLeaderElection,
+		"legacy-leader-elect",
+		false,
+		"Use a legacy leader election method for controller manager instead of the lease API.",
+	)
+
+	flags.Uint8Var(
+		&opts.decryptionConcurrency,
+		"decryption-concurrency",
+		5,
+		"The max number of concurrent policy template decryptions",
+	)
+
+	flags.Uint8Var(
+		&opts.evaluationConcurrency,
+		"evaluation-concurrency",
+		// Set a low default to not add too much load to the Kubernetes API server in resource constrained deployments.
+		2,
+		"The max number of concurrent configuration policy evaluations",
+	)
+
+	flags.BoolVar(
+		&opts.enableMetrics,
+		"enable-metrics",
+		true,
+		"Disable custom metrics collection",
+	)
+
+	flags.Float32Var(
+		&opts.clientQPS,
+		"client-max-qps",
+		30, // 15 * concurrency is recommended
```

Comment:
I was thinking that if the QPS is not explicitly set, then we can do 15 * evaluationConcurrency rather than rely on the user to know they should also adjust QPS.

Reply:
I was planning on doing that kind of configuration in the addon-controller, but I can do it here.